tools/fetch_topsites.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""
+Script for assigning priority to sites.
+
+The rule is as follows:
+    Critical: alexa top 100 in worldwide
+    Important: alexa top 101-1000 in worldwide or alexa top 100 in tier 1 countries/regions
+    Normal: alexa top 1001-10000 or alexa top 101-1000 in tier 1 countries/regions
+"""
+
+import argparse
+
+from tools.topsites.tranco import Tranco
+from tools.topsites.alexa import Alexa
+from tools.topsites.siterank import SiteRank
+from tools.topsites.siterank import DATA_PATH
+from tools.topsites.utils import parse_site_dom_element
+
+REGIONS = ['US', 'FR', 'IN', 'DE', 'TW', 'ID', 'HK', 'SG', 'PL',
+           'GB', 'RU']
+
+
+def main() -> None:
+    description = "Script to fetch and update top site ranking from Tranco and optionally Alexa."
+    parser = argparse.ArgumentParser(description=description)
+
+    parser.add_argument(
+        "--retrieve-regional",
+        action="store_true",
+        help="Whether to retrieve regional ranking from Alexa (access and secret keys will be required).",
+    )
+
+    parser.add_argument(
+        "--ats_access_key",
+        help="AWS access key.",
+        type=str,
+    )
+    parser.add_argument(
+        "--ats_secret_key",
+        help="AWS secret key.",
+        type=str,
+    )
+
+    args = parser.parse_args()
+
+    if args.retrieve_regional:
+        print('Warning: Alexa APIs will be deprecated on December 15, 2022.')
```

Comment:
Should the code have a failsafe feature. Either based on the date. "Hey it's past December 15, 2022. We will ignore the request." Or based on failing the request to Alexa.

Reply:
Thanks, I've added the check 👍